fix(lsp): add request counter and cancellation stub

tris203 opened the pull request:
Fixes #100 
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Enhanced request tracking with per-server request identifiers for clearer operation monitoring.
  * Requests and notifications now provide immediate confirmation, improving reliability of client-server interactions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

neubaner reviewed (commented):
I don't think this methods expects a `function,integer?`. Looking at the docs, `vim.lsp.rpc.PublicClient.request()` should return a `boolean, integer?` and `vim.lsp.rpc.PublicClient.notify()` should return `boolean`.

For reference:
```help
notify({method}, {params})                              *vim.lsp.rpc.notify()*
    Sends a notification to the LSP server.
    ...

    Return: ~
        (`boolean`) `true` if notification could be sent, `false` if not

                                                       *vim.lsp.rpc.request()*
request({method}, {params}, {callback}, {notify_reply_callback})
    Sends a request to the LSP server and runs {callback} upon response.
    ...

    Return (multiple): ~
        (`boolean`) success `true` if request could be sent, `false` if not
        (`integer?`) message_id if request could be sent, `nil` if not

```

tris203 committed:
fix(lsp): add request id tracking and return values

Introduce a local `req_id` counter in the LSP server implementation to
track request identifiers. The `request` method now increments and
returns the current request id along with a success flag. The `notify`
method also returns a success flag for consistency. This change
facilitates better request management and debugging.

neubaner reviewed (approved):
LGTM!
